### ISSUE TITLE
Proxy configuration can't be open anymore with Settings menu if using always incognito mode

### ISSUE THREAD

**Issue by douchesfeedboard** [bromite/bromite#1051]
### Bromite version

Version: `90.0.4430.59`
Arch: `arm64`
Android version: `9.0`
Device model: TA-1033 (Nokia 6 2017)

### Flags

These are the flags changed under `chrome://flags`:

### Is this bug about the SystemWebView?

No

### Is the bug reproducible with latest version?

Yes

### Can the bug be reproduced with corresponding Chromium version?

No (this setting do not exist with Chromium)

### Allow ads for the site; is the bug caused by the adblocker?
No

### Is the bug a crash?

No

### Describe the bug

Proxy configuration can't be open anymore through "Settings" since the latest update 90.0.4430.59 if you have the settings 'Open links in incognito tabs always' activated. You get a white screen for a few seconds before going back to Privacy and security. 
But typing 'chrome://proxy/' manually in the urlbar still working.

### Steps to reproduce the bug

Steps to reproduce the bug:
1. Click on 'the three dots menu in the corner'
2. Click on 'Settings'
3. Click on 'Privacy and security.'
4. Check if 'Open links in incognito tabs always' is on or activate it.
5. Click on 'Proxy configuration'
6. Get a white screen and you're back in the 'Privacy and security' menu

### Expected behavior

A tab with the url 'chrome://proxy/' should be open when clicking on 'Proxy configuration' even when using 'Open links in incognito tabs always'

**Comment by kokukyocho** [comment]
Hello @douchesfeedboard!

I am a bot :robot:

Thanks for submitting this issue!
I noticed that it is missing the template, please edit the issue to match the template for either a [bug](https://github.com/bromite/bromite/blob/master/.github/ISSUE_TEMPLATE/bug_report.md) or a [feature request](https://github.com/bromite/bromite/blob/master/.github/ISSUE_TEMPLATE/feature_request.md).

The issue will be automatically re-opened after that.

**Tip**: if you use GitHub in Desktop mode the template will be proposed to you when submitting issues.

**Comment by csagan5** [comment]
I cannot reproduce this issue; please connect with [remote debugging](https://developer.chrome.com/docs/devtools/remote-debugging/) and check what is the problem on the Console.

**Comment by douchesfeedboard** [comment]
Sorry, I don't have Chrome on my computer for the remote debugging.
But I did some experiments and found the issue is related to the "Open links in incognito tabs always" setting.
When this setting is off, the "Proxy configuration" can be open without issue using the menu in "Privacy and security". 
If the "Open links in incognito tabs always" is on, then the issue occur. I always have this setting on so I'm positive the "Proxy configuration" was working with "Open links in incognito tabs always" prior the actual release.

Can you reproduce the issue if "Open links in incognito tabs always" is on?

**Comment by uazo** [comment]
Current crash is at:

```
04-09 20:19:28.710  4928  4928 W System.err: java.lang.ClassCastException: org.chromium.chrome.browser.customtabs.CustomTabIntentDataProvider cannot be cast to org.chromium.chrome.browser.customtabs.IncognitoCustomTabIntentDataProvider
04-09 20:19:28.710  4928  4928 W System.err:    at org.chromium.chrome.browser.customtabs.CustomTabActivityLifecycleUmaTracker.recordIncognitoLaunchReason(CustomTabActivityLifecycleUmaTracker.java:37)
04-09 20:19:28.711  4928  4928 W System.err:    at org.chromium.chrome.browser.customtabs.CustomTabActivityLifecycleUmaTracker.recordMetrics(CustomTabActivityLifecycleUmaTracker.java:54)
04-09 20:19:28.711  4928  4928 W System.err:    at org.chromium.chrome.browser.customtabs.CustomTabActivityLifecycleUmaTracker.onResumeWithNative(CustomTabActivityLifecycleUmaTracker.java:95)
04-09 20:19:28.711  4928  4928 W System.err:    at org.chromium.chrome.browser.init.ActivityLifecycleDispatcherImpl.dispatchOnResumeWithNative(ActivityLifecycleDispatcherImpl.java:170)
04-09 20:19:28.711  4928  4928 W System.err:    at org.chromium.chrome.browser.init.AsyncInitializationActivity.onResumeWithNative(AsyncInitializationActivity.java:544)
04-09 20:19:28.711  4928  4928 W System.err:    at org.chromium.chrome.browser.app.ChromeActivity.onResumeWithNative(ChromeActivity.java:981)
04-09 20:19:28.711  4928  4928 W System.err:    at org.chromium.chrome.browser.init.NativeInitializationController.onResume(NativeInitializationController.java:166)
04-09 20:19:28.711  4928  4928 W System.err:    at org.chromium.chrome.browser.init.NativeInitializationController.onNativeInitializationComplete(NativeInitializationController.java:144)
04-09 20:19:28.711  4928  4928 W System.err:    at org.chromium.chrome.browser.init.AsyncInitializationActivity.finishNativeInitialization(AsyncInitializationActivity.java:276)
04-09 20:19:28.711  4928  4928 W System.err:    at org.chromium.chrome.browser.app.ChromeActivity.finishNativeInitialization(ChromeActivity.java:1467)
04-09 20:19:28.711  4928  4928 W System.err:    at org.chromium.chrome.browser.customtabs.BaseCustomTabActivity.finishNativeInitialization(BaseCustomTabActivity.java:294)
04-09 20:19:28.711  4928  4928 W System.err:    at org.chromium.chrome.browser.customtabs.CustomTabActivity.finishNativeInitialization(CustomTabActivity.java:150)
04-09 20:19:28.711  4928  4928 W System.err:    at org.chromium.chrome.browser.init.BrowserParts$$CC.startNativeInitialization$$dflt$$(Unknown Source:63)
04-09 20:19:28.711  4928  4928 W System.err:    at org.chromium.chrome.browser.init.AsyncInitializationActivity.startNativeInitialization(Unknown Source:0)
04-09 20:19:28.711  4928  4928 W System.err:    at org.chromium.chrome.browser.init.ChromeBrowserInitializer.lambda$handlePostNativeStartup$6$ChromeBrowserInitializer(ChromeBrowserInitializer.java:279)
04-09 20:19:28.711  4928  4928 W System.err:    at org.chromium.chrome.browser.init.ChromeBrowserInitializer$$Lambda$7.run(Unknown Source:2)
04-09 20:19:28.711  4928  4928 W System.err:    at org.chromium.chrome.browser.init.ChainedTasks$1.run(ChainedTasks.java:37)
04-09 20:19:28.711  4928  4928 W System.err:    at android.os.MessageQueue.nativePollOnce(Native Method)
04-09 20:19:28.711  4928  4928 W System.err:    at android.os.MessageQueue.next(MessageQueue.java:326)
04-09 20:19:28.711  4928  4928 W System.err:    at android.os.Looper.loop(Looper.java:160)
04-09 20:19:28.711  4928  4928 W System.err:    at android.app.ActivityThread.main(ActivityThread.java:6669)
04-09 20:19:28.711  4928  4928 W System.err:    at java.lang.reflect.Method.invoke(Native Method)
04-09 20:19:28.711  4928  4928 W System.err:    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
04-09 20:19:28.711  4928  4928 W System.err:    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
04-09 20:19:28.722  5033  5033 D cr_LibraryLoader: [LibraryLoader.java:320] Configuration, useChromiumLinker = true, useModernLinker = false
```

**Comment by csagan5** [comment]
@uazo so the proxy page is a custom tab intent and it crashes while attempting to convert it?

**Comment by uazo** [comment]
> so the proxy page is a custom tab intent

yes, like all chrome:// pages.

> and it crashes while attempting to convert it?

I think incognito customtabs are something new in v90, probably before native pages (like proxy ui) were always launched as non incognito custom pages.
I think the patch to change is the always incognito, because it should launch native internal pages as normal pages, like before v90.
I haven't checked yet but probably the other chrome:// pages have the same problem as well.

note: maybe we would do better to add in the issue template if the always incognito mode is active.

**Comment by douchesfeedboard** [comment]
> note: maybe we would do better to add in the issue template if the always incognito mode is active.

I did the change in the template and title.

**Comment by csagan5** [comment]
Fixed in `90.0.4430.74`.
